connect/runtime/src/main/java/org/apache/kafka/connect/util/InstantiableClassValidator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.util;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.utils.Utils;
+
+public class InstantiableClassValidator implements ConfigDef.Validator {
+
+    @Override
+    public void ensureValid(String name, Object value) {
+        if (value == null) {
+            // The value will be null if the class couldn't be found; no point in performing follow-up validation
+            return;
+        }
+
+        Class<?> cls = (Class<?>) value;
+        try {
+            Object o = cls.getDeclaredConstructor().newInstance();
+            Utils.maybeCloseQuietly(o, o + " (instantiated for preflight validation");
+        } catch (NoSuchMethodException e) {
+            throw new ConfigException(name, cls.getName(), "Could not find a public no-argument constructor for class" + (e.getMessage() != null ? ": " + e.getMessage() : ""));
+        } catch (ReflectiveOperationException | RuntimeException e) {
+            throw new ConfigException(name, cls.getName(), "Could not instantiate class" + (e.getMessage() != null ? ": " + e.getMessage() : ""));
+        }
```

Comment:
Could we leverage [Utils::newInstance](https://github.com/apache/kafka/blob/dbda60c60da8f5a7eabe113615196b729b40a0e8/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L390-L403) here?

Reply:
Ah, good idea but it swallows the two errors we're trying to catch and wrap here and throws them as `KafkaException` instead of `ConfigException`, which we definitely want because of how they're [treated specially](https://github.com/apache/kafka/blob/c2bb8eb875d94568d1ad19bf207ec69c182405d4/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L634-L636) during config validation.

Reply:
Right, I was actually thinking about handling that `KafkaException` and throwing a `ConfigException` with the same message. Although now that I think about it some more, it doesn't sound much better / cleaner than the current approach anyway. I'm fine with either approach 👍

Reply:
Considering the duplication is minor and it sets us up for flexibility in the future if we choose to expand our coverage for error messages, I'd prefer to keep this as-is for now.